core/src/display_object/edit_text.rs

```diff
@@ -872,8 +883,22 @@ impl<'gc> EditText<'gc> {
                             context.renderer.draw_rect(color.clone(), &caret);
                         }
                     }
-                },
-            );
+                };
+            if edit_text.is_password {
+                font.evaluate(
+                    &"*".repeat(text.len()),
```

Comment:
This uses the normal text's layout when rendering the asterisks, which can be different. Think of "WWWWWWWW" vs. the displayed password "********", which will word-wrap at different points in a multiline text box. So I think the password replacement should happen elsewhere, in `relayout`, and the characters can be swapped at that point. Ideally we could find a way to avoid the per-frame allocation that `str::repeat` does, but I'm not too worried about that for a first pass.

Reply:
Your definitely right, the reason I did that was because I was worried if I replaced the actual text (like in text_spans) I would overwrite the actual value of the input field, ill try doing the replacement in relayout like you said.